### PR TITLE
add prototype of distributed inference computing with Seldon & Ray

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

Example on using Ray to distribute single request as mini-batches. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Fri Oct  2 21:39:57 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=1

**Comment by seldondev** [comment]
Fri Oct  2 21:51:17 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=3

**Comment by seldondev** [comment]
Fri Oct  2 21:51:18 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=4

**Comment by seldondev** [comment]
Fri Oct  2 21:53:32 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=2

**Comment by seldondev** [comment]
Wed Oct 14 15:05:51 UTC 2020
The logs for [integration] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=5

**Comment by seldondev** [comment]
Mon Nov  9 11:17:14 UTC 2020
The logs for [pr-build] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=6

**Comment by seldondev** [comment]
Mon Nov  9 11:17:25 UTC 2020
The logs for [lint] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=7

File: examples/ray/batch-split-proxy/Dockerfile
Comment (axsaucedo) [inline examples/ray/batch-split-proxy/Dockerfile:2]: update to 1.5

File: examples/ray/batch-split-proxy/training/requirements.txt
Comment (axsaucedo) [inline examples/ray/batch-split-proxy/training/requirements.txt:1]: I believe the current master uses 0.5.5

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Added a couple of small comments

File: examples/ray/batch-split-proxy/training/requirements.txt
Comment (RafalSkolasinski) [reply examples/ray/batch-split-proxy/training/requirements.txt:1]: This is only used to download training dataset.

File: examples/ray/batch-split-proxy/Dockerfile
Comment (RafalSkolasinski) [reply examples/ray/batch-split-proxy/Dockerfile:2]: 1.5.0-dev?

Comment (axsaucedo) [reply examples/ray/batch-split-proxy/Dockerfile:2]: yeah

Comment (RafalSkolasinski) [reply examples/ray/batch-split-proxy/Dockerfile:2]: Don't expect this to be frequently rebuild so I'd rather go with stable base, `1.4.0` maybe?

File: examples/ray/batch-split-proxy/training/requirements.txt
Comment (axsaucedo) [reply examples/ray/batch-split-proxy/training/requirements.txt:1]: Ok should be fine

File: examples/ray/batch-split-proxy/Dockerfile
Comment (axsaucedo) [reply examples/ray/batch-split-proxy/Dockerfile:2]: This shoudl be updated by the release script, so it should be in-line with the version in version.txt

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Nov  9 17:11:49 UTC 2020
The logs for [pr-build] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=8

**Comment by seldondev** [comment]
Mon Nov  9 17:11:53 UTC 2020
The logs for [lint] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=9

File: examples/ray/batch-split-proxy/Dockerfile
Comment (RafalSkolasinski) [reply examples/ray/batch-split-proxy/Dockerfile]: Fixed

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 6f56653ede6c6577c26a9dffd36a9b50fb710c4e | [link]() | `/test integration`
pr-build | acbd0d95cafd37e73e232993f84e758250930cf8 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
There was some pr-build failure, looked like lint / licence related, I re-based onto master now.

**Comment by seldondev** [comment]
Mon Nov  9 17:27:55 UTC 2020
The logs for [lint] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=11

**Comment by seldondev** [comment]
Mon Nov  9 17:28:22 UTC 2020
The logs for [pr-build] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=10

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/2517#issuecomment-724696226 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-724696226/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Nov 10 13:21:29 UTC 2020
The logs for [lint] [13] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/13.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=13

**Comment by seldondev** [comment]
Tue Nov 10 13:21:43 UTC 2020
The logs for [pr-build] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2517/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2517 --build=12

**Pull request merged**
